### PR TITLE
[Enhancement](Column) Support int default value to int type column

### LINKED ISSUE #20670: [Enhancement] default value for bigint should not has to be quoted
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

CREATE TABLE t (topic BIGINT DEFAULT 0) does not work. CREATE TABLE t (topic BIGINT DEFAULT "0") works. 

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by waywtdcc**
## Proposed changes

Issue Number: close  #20670 #24125

## Further comments

Support int/bigint/decimal default value to int/bigint/decimal type column

### CODE REVIEW

**Comment by waywtdcc** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.86 seconds
 stream load tsv:          536 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17161920510 Bytes

**Comment by waywtdcc** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.99 seconds
 stream load tsv:          540 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162085244 Bytes

**Review by zy-kkk — COMMENTED** [review]
You should add new tests, including create, alter, insert into, stream load, and not modify the old ones. We need to ensure compatibility with the old behaviors

**Comment by waywtdcc** [comment]
> You should add new tests, including create, alter, insert into, stream load, and not modify the old ones. We need to ensure compatibility with the old behaviors

Thanks for your review. I made a commit, is that right?

**Comment by waywtdcc** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.33 seconds
 stream load tsv:          530 seconds loaded 74807831229 Bytes, about 134 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162058782 Bytes

File: regression-test/suites/load_p0/insert/test_insert_default_value.groovy [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    """
+    
+    sql """ set enable_nereids_planner=true """
+    sql """ set enable_nereids_dml=true """
+    sql """ insert into ${insert_tbl_dft} values() """
+
+    sql """ set enable_nereids_planner=false """
+    sql """ set enable_nereids_dml=false """
+    sql """ insert into ${insert_tbl_dft} values() """
+
```
Comment (TangSiyang2001) [inline regression-test/suites/load_p0/insert/test_insert_default_value.groovy:89]: Please add test case for stream load.

**Comment by waywtdcc** [comment]
run buildall

**Comment by waywtdcc** [comment]
@TangSiyang2001  @zy-kkk  hello,can you help me to review this pr?

**Review by Hastyshell — APPROVED** [review]
LGTM

**Review by zy-kkk — DISMISSED** [review]
LGTM

**Comment by zy-kkk** [comment]
run buildall

**Comment by waywtdcc** [comment]
Hi everyone, any questions? Can it be merged?

**Comment by waywtdcc** [comment]
@TangSiyang2001 hello?Can this pr merge to master?

**Review by zy-kkk — APPROVED** [review]
LGTM

**Comment by waywtdcc** [comment]
this pr can merged?

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
@waywtdcc Sorry for late reply, there are conflicts need to be resolved.

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 2]
```diff
@@ -3671,6 +3671,18 @@ opt_default_value ::=
     {:
         RESULT = ColumnDef.DefaultValue.currentTimeStampDefaultValueWithPrecision(precision);
     :}
+    | KW_DEFAULT INTEGER_LITERAL:value
+    {:
+        RESULT = new ColumnDef.DefaultValue(true, value);
```
Comment (morrySnow) [inline fe/fe-core/src/main/cup/sql_parser.cup:3676]: update Nereids' parser and CreateTableCommand too

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Pull request closed without merging**